python/cudf/cudf/core/dataframe.py

```diff
@@ -6864,6 +6816,17 @@ def unstack(self, level=-1, fill_value=None):
             self, level=level, fill_value=fill_value
         )
 
+    def equals(self, other):
+        frames_equal = super().equals(other)
+        if frames_equal:
+            # additionally, check equality of names
+            for self_name, other_name in zip(
+                self._data.names, other._data.names
+            ):
+                if self_name != other_name:
+                    return False
```

Comment:
We should check the names / dtypes first so we can short circuit the expensive equality check

Reply:
Fixed